packages/blade/src/components/Input/_decisions/decisions.md

```diff
@@ -0,0 +1,212 @@
+  showClearButton
+  helpText="Enter first name and last name. Eg: Kamlesh Chandnani"
+  errorText="Name cannot be blank"
+  successText="Name validated"
+  validationState={inputValue.length < 10 ? 'error' : 'success'}
+/>
+```
+
+The mobile keyboard type will be automatically determined based on the `type` attribute. Check out [Web & React Native Attributes for showing keyboard types and autocomplete suggestions based on input](#web--react-native-attributes-for-showing-keyboard-types-and-autocomplete-suggestions-based-on-input) section for more details
+
+| Prop                                                                                               | Type                                                                      | Required                                                                            | Default                                                   | Description                                                                                                                                                                                                                                                                           |
+| -------------------------------------------------------------------------------------------------- | ------------------------------------------------------------------------- | ----------------------------------------------------------------------------------- | --------------------------------------------------------- | ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| type                                                                                               | `text`, `search`, `tel`, `email`, `url`, `numeric`                        | No                                                                                  | `text`                                                    | type of text to be rendered. this will also determine the keyboard type to be show on mobile devices                                                                                                                                                                                  |
+| label                                                                                              | `string`                                                                  | (confirm if we have a use case for input field without labels(search field maybe?)) |                                                           | Determines the label of an input field. Also used as `aria-label`                                                                                                                                                                                                                     |
```

Comment:
I think we should make `label` a required field for `TextInput`. Design systems generally have a different component like Search or QueryField to be used for the SearchField. It's better to keep them separate as they serve different purposes and most likely the SearchField will have additional props. In my previous experience, we had to learn from our mistakes and then built it out separately [here](https://design.gs.com/components/search). 😄

Reply:
On second thoughts, I can see consumers would want to use the input fields in conjunction with data tables or as filters where they may want to skip the label. Could make it optional, entirely depends on the guidelines we would want to set.

Reply:
making it required for now. once we have a use case for making it optional we can add that as an enhancement. changing required to optional is an easy migration path but reverse is not true and scary 😄